api/src/backend/tasks/jobs/deletion.py

```diff
@@ -21,33 +21,36 @@ def delete_provider(pk: str):
 
     Raises:
         Provider.DoesNotExist: If no instance with the provided primary key exists.
+        DatabaseError: If any deletion step fails.
     """
     instance = Provider.all_objects.get(pk=pk)
+
     deletion_summary = {}
 
-    with transaction.atomic():
-        # Delete Scan Summaries
-        scan_summaries_qs = ScanSummary.all_objects.filter(scan__provider=instance)
-        _, scans_summ_summary = batch_delete(scan_summaries_qs)
-        deletion_summary.update(scans_summ_summary)
-
-        # Delete Findings
-        findings_qs = Finding.all_objects.filter(scan__provider=instance)
-        _, findings_summary = batch_delete(findings_qs)
-        deletion_summary.update(findings_summary)
-
-        # Delete Resources
-        resources_qs = Resource.all_objects.filter(provider=instance)
-        _, resources_summary = batch_delete(resources_qs)
-        deletion_summary.update(resources_summary)
-
-        # Delete Scans
-        scans_qs = Scan.all_objects.filter(provider=instance)
-        _, scans_summary = batch_delete(scans_qs)
-        deletion_summary.update(scans_summary)
-
-        provider_deleted_count, provider_summary = instance.delete()
-        deletion_summary.update(provider_summary)
+    deletion_steps = [
+        ("Scan Summaries", ScanSummary.all_objects.filter(scan__provider=instance)),
+        ("Findings", Finding.all_objects.filter(scan__provider=instance)),
+        ("Resources", Resource.all_objects.filter(provider=instance)),
+        ("Scans", Scan.all_objects.filter(provider=instance)),
+    ]
+
+    for step_name, queryset in deletion_steps:
+        try:
+            with transaction.atomic():
+                _, step_summary = batch_delete(queryset)
+                deletion_summary.update(step_summary)
+        except DatabaseError as error:
+            logger.error(f"Error deleting {step_name}: {error}")
+            raise
```

Comment:
```suggestion
            raise error
```

Reply:
Not needed. Python will raise the captured exception by default.